Feat: Make desktop app landing page optional (#189)
mattermost/mattermost-test-management: data/test-cases/desktop-app--native-specific-/landing-page/MM-T5640.md

---
# (Required) Ensure all values are filled up
name: "Verify user is not taken to landing page when EnableDesktopLandingPage is enabled"
status: Active
priority: Normal
folder: Landing Page
authors: "@yasserfaraazkhan"
team_ownership: 
- ICU
priority_p1_to_p4: P2 - Core Functions (Do core functions work?)

# (Optional)
location: Smoke Test
component: null
tags: []
labels: []
tested_by_contributor: ""

# (Optional) Test type and tools
cypress: N/A
detox: N/A
mmctl: N/A
playwright: N/A
rainforest: []
manual_test_environments: []

# Do not change
id: 193404883
key: MM-T5640
created_on: "2024-10-17T20:07:08Z"
last_updated: "2024-10-18T01:37:07Z"
case_hashed: 2fdb965026d4df14f5510f39f8d5de574a7679899ddb353dc221fbe8cc9642cbc4166ef3836043d46f4b3861f251184c
steps_hashed: 85cd742dfa69034f94a01178c3bd4481ee32571ada47319703f4de08652fffa7a9fba25d191ff7b3d6b0c4db23cbd163
---

<!-- (Auto-generated) Based on frontmatter's "key" and "name" -->

## MM-T5640: Verify user is not taken to landing page when EnableDesktopLandingPage is enabled

---

**Step 1**

> Scenario 1: Disable `ServiceSettings.EnableDesktopLandingPage` and verify user is not taken to landing page.

Reviewer: ```suggestion
Scenario: Disable `ServiceSettings.EnableDesktopLandingPage` and verify user is not taken to landing page.
```

Steps:

1. Disable the setting in Config and start the server.
2. Visit the server URL.

> 3. Veirfy user is not redirected to landing page

Reviewer: ```suggestion
3. Verify user is not redirected to landing page
```

> 
> Scenario 1: Verify the by `ServiceSettings.EnableDesktopLandingPage` is enabled.

Reviewer: If this is step 2 and is expected for MM-T5640_2, then

```suggestion
---

**Step 2**

Scenario: Enable `ServiceSettings.EnableDesktopLandingPage` and verify user is taken to landing page.
```

Steps:

1. Start the server.
2. Visit the server URL.

> 3. Veirfy user is redirected to landing page.

Reviewer: ```suggestion
3. Verify user is redirected to landing page.
```
